### PR TITLE
*: reduce NewRuntimeStatsColl() object allocation

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Take part of the code from https://github.com/pingcap/tidb/pull/26007
ResetContextStmt is not touched

### What is changed and how it works?

What's Changed:

How it Works:

Change `NewRuntimeStatsColl()` to `NewRuntimeStatsColl(*RuntimeStatsColl)` so when the argument is not nil, the object is reused.

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

- No release note

### CODE REVIEW

**Comment by tiancaiamao** [comment]
For https://github.com/pingcap/tidb/issues/25573

**Review by xhebox — APPROVED** [review]

File: util/execdetails/execdetails.go [hunk 1]
```diff
@@ -555,9 +555,24 @@ type RuntimeStatsColl struct {
...
-func NewRuntimeStatsColl() *RuntimeStatsColl {
-	return &RuntimeStatsColl{rootStats: make(map[int]*RootRuntimeStats),
-		copStats: make(map[int]*CopRuntimeStats)}
+// Reuse the object to reduce allocation when *RuntimeStatsColl is not nil.
+func NewRuntimeStatsColl(reuse *RuntimeStatsColl) *RuntimeStatsColl {
+	if reuse != nil {
+		// Reuse map is cheaper than create a new map object.
+		// Go compiler optimize this cleanup code pattern to a clearmap() function.
+		for k := range reuse.rootStats {
+
```
Comment (crazycs520) [inline util/execdetails/execdetails.go:564]: remove this blank line.

Comment (tiancaiamao) [reply util/execdetails/execdetails.go]: Done

**Review by crazycs520 — APPROVED** [review]
LGTM

**Review by tangenta — APPROVED** [review]

**Pull request merged**